clients/pom.xml

```diff
@@ -44,10 +44,19 @@
 	  <plugin>
 	    <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-gpg-plugin</artifactId>
-	    <version>1.5</version> <!-- must path java/pom.xml -->
+	    <version>1.5</version> <!-- must match java/pom.xml -->
 	    <configuration>
 	      <keyname>A670BB5ACA82B971772B6DEFDFFC0764F5A9A357</keyname>
 	    </configuration>
```

Comment:
given that you created the github workflow to publish to maven central, is the keyname still needed in this pom? (I was not sure about the answer by just reading the docs)

Reply:
Yes.

The secrets `OSSRH_USERNAME` and `OSSRH_TOKEN` are to authenticate to OSSRH Sonatype Nexus, which is the repository for Maven Central.  But Maven Central itself _also_ requires that all artifacts be signed, using a GPG key.  That GPG secret key is inserted through secret `OSSRH_GPG_SECRET_KEY`; I inserted it into the GitHub secret environment "Treeverse signing".  This keyname is the ID of the signing key; this configuration means that (by default) we sign with our key.  (You can deploy from a developer laptop by installing and accessing this GPG secret key, of course).

One confusing thing is that  two identities are involved:
1. User on OSSRH Sonatype Nexus;
2. JAR signer.
Indeed AFAICT this is a security bug in Maven Central: AFAICT there is no binding between the two identities.  So JAR signing is required, but all JARs might as well be signed by Bugs Bunny.
![JARs might as well be signed by Bugs Bunny: "What's up, doc?"](https://user-images.githubusercontent.com/7802932/117113236-dd9fcd00-ad92-11eb-90ad-895d779be1b2.png)

Reply:
Thanks for the thorough explanation!